scripts/fiximports/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"golang.org/x/tools/imports"
+)
+
+var (
+	// groupByPrefixes is the list of import prefixes that should _each_ be grouped separately.
+	// See: imports.LocalPrefix.
+	groupByPrefixes = []string{
+		"github.com/filecoin-project",
+		"github.com/filecoin-project/lotus",
+	}
+	newline                  = []byte("\n")
+	importBlockRegex         = regexp.MustCompile(`(?s)import\s*\((.*?)\)`)
+	consecutiveNewlinesRegex = regexp.MustCompile(`\n\s*\n`)
+)
+
+func main() {
+	if err := filepath.Walk(".", func(path string, info fs.FileInfo, err error) error {
+		switch {
+		case err != nil:
+			return err
+		case // Skip ffi and test vectors in "./extern" directory and their contents.
+			strings.HasSuffix(path, "extern/filecoin-ffi"),
+			strings.HasSuffix(path, "extern/test-vectors"):
+			return filepath.SkipDir
+		case // Skip directories, generated cborgen go files and any other non-go files.
+			info.IsDir(),
+			strings.HasSuffix(info.Name(), "_cbor_gen.go"),
+			!strings.HasSuffix(info.Name(), ".go"):
+			return nil
+		}
+		return fixGoImports(path)
```

Comment:
I'd spin this up in a goroutine, should make the whole thing much fasterer

Reply:
Yep I have tried that with errgroups. it's about 4x faster than current 👍

Reply:
Sigh... goimports programatic API is not thread safe unfortunately.

Reply:
Wow, that's a nasty API... We could do it in two steps (this time, we probably would close the files after reading them), but that's probably more trouble than it's worth for now.

Reply:
yep. I think the thing to do is to simplify our formatting to two groups of SDK and no SDK, and frankly use another tool to sort imports.  `goimports` seems very slow in addressing the feature requests.